NGit/NGit.Patch/FileHeader.cs

```diff
@@ -711,6 +715,10 @@ private static string P1(string r)
 		internal virtual FileMode ParseFileMode(int ptr, int end)
 		{
 			int tmp = 0;
+		    if (buf[end - 2] == '\r') // Ignore windows line ending
```

Comment:
I suspect this is a way to detect that the line ending is \r\r\n. I would probably extract this in a method and test all the three characters (with overflow checks) and also reuse it in the other places where you do buf[dot2 - 2] == '\r'